serving/src/main/java/feast/serving/interceptors/GrpcMonitoringInterceptor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.serving.interceptors;
+
+import feast.serving.util.Metrics;
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+
+/**
+ * GrpcMonitoringInterceptor intercepts GRPC calls to provide request latency histogram metrics in
+ * the Prometheus client.
+ */
+public class GrpcMonitoringInterceptor implements ServerInterceptor {
```

Comment:
Bikeshedding on naming if you'll bear with me… it's more about request/response metrics than monitoring, monitoring would be what you might externally do with the metrics ☺️

Also I wonder if anyone would object to dropping the `Grpc` prefix, I imagine everything in the `interceptors` package will be a gRPC interceptor so perhaps it's redundant. What do you think of:

```suggestion
public class RequestMetricsInterceptor implements ServerInterceptor {
```

Reply:
Yes sounds reasonable to me. I think we need to create a new PR?

Although the sample library for integrating Prometheus with GRPC in Java uses the monitoring term as well :)
https://github.com/grpc-ecosystem/java-grpc-prometheus/tree/master/src/main/java/me/dinowernli/grpc/prometheus

I guess as long as we're consistent.